io/concatenateHDF5.m

```diff
@@ -0,0 +1,157 @@
+% hdf5Name = 'test.hdf5';
+% downsmpFactor = 0.5;
+% plusmazeName = 'mouse5_day20_ego-right.txt';
+% trim = [10,5];
+%
+% 2015-01-27 Jessica Maxey
+
+outputFileName = fullfile(outputDir,[plusmazeName(1:end-4),'_new.txt']);
+outputFile = fopen(outputFileName,'wt+');
+
+[frame_indices,location_info,time] = getTrialIndicies(fullfile(tifDir,plusmazeName));
+startFrames = frame_indices(:,1);
+
+list = dir(fullfile(tifDir,'*.tif'));
+firstName = fullfile(tifDir,list(1).name);
+[rows,cols] = size(imread(firstName));
+numFiles = length(list);
+
+totalFrames = 1;
+
+downsmpRows = floor(rows*downsmpFactor);
+downsmpCols = floor(cols*downsmpFactor);
+chunkSize = [downsmpRows downsmpCols 1];
+dataset = '/Data/Images';
+h5create(fullfile(outputDir,hdf5Name),dataset,[Inf Inf Inf],'ChunkSize',chunkSize,'Datatype','uint16');
+badTrial = 0;
+skipCount = 0;
+testFrames = 1;
+trialCount = 0;
+
+for i=1:length(list)
+    if(i ~= 1)
+        %%% Find if the start frame of the next recording matches one of
+        %%% the start frame indicies
+        idx = find(testFrames == startFrames);
+        if(isempty(idx))
+            %%% Bad trial
+            badTrial = 1;
+            tif_info = imfinfo(fullfile(tifDir,list(i).name));
+            numBadFrames = length(tif_info);
+            testFrames = testFrames + numBadFrames;
+        else
+            %%% Good trial
+            badTrial = 0;
+        end
+    end
+
+    if(badTrial == 0)
+        %%% Add the good trial to the hdf5 file
+        trialCount = trialCount+1;
+        tifName = fullfile(tifDir,list(i).name);
+        tifInfo = imfinfo(tifName);
+        tifFile = Tiff(tifName,'r');
+        oriFrames = length(tifInfo);
+        
+        xmlName = [tifName(1:end-3),'xml'];
+        xmlData = parse_miniscope_xml(xmlName);
+        
+        %%% Determine if frames were dropped by the miniscope during the
+        %%% trial
+        if(str2num(xmlData.dropped_count) ~= 0)
+            droppedFrames = str2num(xmlData.dropped);
+        else
+            numFrames = oriFrames;
+            droppedFrames = 0;
+        end
+        
+        %%% Read in the images, trim the frames at the beginning and end 
+        %%% of the trial (set by the 'trim' argument), and downsample by the 
+        %%% 'downsmpFactor' argument
+        imageStack = uint16(zeros(floor(rows*downsmpFactor),floor(cols*downsmpFactor),oriFrames-trim(1)-trim(2)));
+        if(downsmpFactor == 1)
+            for j=1:oriFrames-trim(1)-trim(2)
+                tifFile.setDirectory(j+trim(1));
+                imageStack(:,:,j) = uint16(tifFile.read());
+            end
+        else
+            for j=1:oriFrames-trim(1)-trim(2)
+                tifFile.setDirectory(j+trim(1));
+                imageStack(:,:,j) = uint16(imresize(tifFile.read(),downsmpFactor,'bilinear'));
+            end
+        end
+        
+        %%% Replace any dropped frames with the frame immediately
+        %%% preceeding it

```

Comment:
I wouldn't worry about it for now. Maybe just write a `TODO: ...` in the comments.